lib/procto.rb

```diff
@@ -77,6 +78,12 @@ def included(host)
       define_singleton_method(:call, &block)
     end
 
+    if @name
```

Comment:
Only thing I would change is to move this inside the other instance_exec class. with a conditional:

```rb
define_singleton_method(method_name, &block) if method_name
```